contracts/src/SP1ICS07Tendermint.sol

```diff
@@ -421,33 +433,41 @@ contract SP1ICS07Tendermint is
     }
 
     /// @notice Validates the Env public values.
-    /// @param env The public values.
-    function validateEnv(Env memory env) private view {
-        if (env.now > block.timestamp) {
-            revert ProofIsInTheFuture(block.timestamp, env.now);
+    /// @param publicClientState The public client state.
+    /// @param time The time.
+    function validateEnv(ClientState memory publicClientState, uint64 time) private view {
+        if (time > block.timestamp) {
+            revert ProofIsInTheFuture(block.timestamp, time);
         }
-        if (block.timestamp - env.now > ALLOWED_SP1_CLOCK_DRIFT) {
-            revert ProofIsTooOld(block.timestamp, env.now);
+        if (block.timestamp - time > ALLOWED_SP1_CLOCK_DRIFT) {
+            revert ProofIsTooOld(block.timestamp, time);
         }
-        if (keccak256(bytes(env.chainId)) != keccak256(bytes(clientState.chainId))) {
-            revert ChainIdMismatch(clientState.chainId, env.chainId);
+
+        // NOTE: We do not check the equality of latest height and isFrozen
+        if (keccak256(bytes(publicClientState.chainId)) != keccak256(bytes(clientState.chainId))) {
+            revert ChainIdMismatch(clientState.chainId, publicClientState.chainId);
         }
-        if (
-            env.trustThreshold.numerator != clientState.trustLevel.numerator
-                || env.trustThreshold.denominator != clientState.trustLevel.denominator
-        ) {
+        if (publicClientState.trustLevel.numerator != clientState.trustLevel.numerator) {
+            revert TrustThresholdMismatch(
+                clientState.trustLevel.numerator,
+                clientState.trustLevel.denominator,
+                publicClientState.trustLevel.numerator,
+                publicClientState.trustLevel.denominator
+            );
+        }
+        if (publicClientState.trustLevel.denominator != clientState.trustLevel.denominator) {
             revert TrustThresholdMismatch(
                 clientState.trustLevel.numerator,
                 clientState.trustLevel.denominator,
-                env.trustThreshold.numerator,
-                env.trustThreshold.denominator
+                publicClientState.trustLevel.numerator,
+                publicClientState.trustLevel.denominator
             );
         }
-        if (env.trustingPeriod != clientState.trustingPeriod) {
-            revert TrustingPeriodMismatch(clientState.trustingPeriod, env.trustingPeriod);
+        if (publicClientState.trustingPeriod != clientState.trustingPeriod) {
+            revert TrustingPeriodMismatch(clientState.trustingPeriod, publicClientState.trustingPeriod);
         }
-        if (env.trustingPeriod > clientState.unbondingPeriod) {
```

Comment:
Why was this removed? Would we catch this anywhere else now?

Reply:
I think this would be handled in the constructor now since you can't even create a client in this manner

Reply:
That would make sense. Do we have a test that covers that scenario (if not, we should create an issue to do that)?

Reply:
We don't have any constructor tests, so perhaps an issue for that?

Reply:
Added an issue: https://github.com/cosmos/sp1-ics07-tendermint/issues/129